### PR TITLE
New Zerg Speedrun Challenges

### PR DESCRIPTION
**Pull request by Gemster312**
## What's Added/Changed?
3 new locations have been added.

* 1 New Speedrun Location in **HotS - Lab Rat**
> Location: "Lab Rat: Win In Under 10 Minutes"
* 1 New Speedrun Location in **HotS - Rendezvous**
> Location: "Rendezvous: Kill All Before Reinforcements"
* 1 New Speedrun Location in **HotS - The Reckoning**
> Location: "The Reckoning: Trash the Odin Early"

## Tests Completed for each new location
See paired Archipelago-SC2-data PR

## Pairing Archipelago-SC2-data PR
[PR-206](https://github.com/Ziktofel/Archipelago-SC2-data/pull/206)

### CODE REVIEW

File: worlds/sc2/locations.py [hunk 1]
```diff
@@ -1099,6 +1105,8 @@ def get_locations(world: Optional['SC2World']) -> Tuple[LocationData, ...]:
                      lambda state: logic.the_reckoning_requirement(state)),
         make_location_data(SC2Mission.THE_RECKONING.mission_name, "Odin", SC2HOTS_LOC_ID_OFFSET + 2004, LocationType.EXTRA,
                      lambda state: logic.the_reckoning_requirement(state)),
+        make_location_data(SC2Mission.THE_RECKONING.mission_name, "Trash the Odin Early", SC2HOTS_LOC_ID_OFFSET + 2005, LocationType.SPEEDRUN,
```
Comment (MatthewMarinets) [inline worlds/sc2/locations.py:1108]: Would be nice if these location names still gave a time limit in minutes / seconds.

Comment (Gemster312) [reply worlds/sc2/locations.py:1108]: Unfortunately the Odin gets sent at different times depending on difficulty.
Plus, it's not on a global timer, so there is a bit of variability depending on if you manage to skip certain dialogues at the start of the mission/how soon you spawn in Jim Raynor.

Comment (Ziktofel) [reply worlds/sc2/locations.py:1108]: Can unupgraded Roach + Hydra do it with banana economy? Maybe at least one of Defiler or Infestor are needed

Comment (Gemster312) [reply worlds/sc2/locations.py:1108]: Honestly not sure. I believe on Brutal difficulty you get around 22.5 minutes to complete this objective. Given that `logic.the_reckoning_requirement` is equivalent to `zerg_competent_comp and zerg_competent_anti_air` I think this should be fine, but if others have different suggestions feel free to post what I should replace it with.

**Pull request merged**